### ISSUE TITLE
      Stop On Exit doesn't work

### ISSUE THREAD

**Issue by Fargus-Van-Crystal-de-Mindark** [Epicnicity322/PlayMoreSounds#3]
Hello

I tried to set the stop on exit in region but it doesn't work

```
# Set a sound to play when you enter, exit or stand on a specific region.
#
# Sample:
# (Take a note that this is a sample and the sounds may not be available
# on your MC version.)
#
#PlayMoreSounds: # The region plugin.
#  Spawn: # The region name, replace 'Spawn' with the name of your region.
#    Enter: # When a player enters this region.
#      Cancellable: true
#      Enabled: true
#      Stop On Exit:
#        Enabled: true # If enabled, the sound will be stopped when the player leaves the region.
#        Delay: 20 # The time to wait before stopping the sound.
#      Prevent Default Sound: true # If enabled, Region Enter sound in sounds.yml won't be played.
#      Sounds:
#        '0':
#          Delay: 0
#          Options:
#            Radius: 0
#          Pitch: 1
#          Sound: BLOCK_NOTE_BLOCK_PLING
#          Volume: 10
#    Leave: # When a player exits this region.
#      Cancellable: true
#      Enabled: true
#      Prevent Default Sound: true # If enabled, Region Leave sound in sounds.yml won't be played.
#      Sounds:
#        '0':
#          Delay: 0
#          Options:
#            Radius: 0
#          Pitch: 1
#          Sound: BLOCK_NOTE_BLOCK_BASS
#          Volume: 10
#    Loop: # When a player enters the region, a loop will be triggered and play.
#      Cancellable: true
#      Delay: 0 # Time in ticks to wait to start the loop once the player enters the region.
#      Enabled: true
#      Period: 100 # Time in tick the loop will wait until playing the sound again.
#      # If you have a long song playing, when the player leaves this region, the song
#      #will be stopped instead of playing until the end. This setting applies to sounds only,
#      #the loop function is stopped automatically.
#      Stop On Exit:
#        Delay: 10
#        Enabled: true
#      Prevent Other Sounds:
#        Enter Sound: true # If enabled, Enter sound in regions.yml won't be played.
#        Default Sound: true # If enabled, Region Enter sound in sounds.yml won't be played.
#      Sounds:
#        '0':
#          Delay: 0
#          Options:
#            Radius: 0
#          Pitch: 1
#          Sound: BLOCK_NOTE_BLOCK_BASS
#          Volume: 10
#
# You can only play sounds in PMS native regions. To play to other plugins, search for compatibility
#addons on https://www.spigotmc.org/resources/37429/
#
# More information about sounds on sounds.yml.

Version: 3.3.0

# Region Sounds
PlayMoreSounds: # The region plugin.
  TEST: # The region name, replace 'Spawn' with the name of your region.
    Enter: # When a player enters this region.
      Cancellable: true
      Enabled: true
      Stop On Exit:
        Enabled: true # If enabled, the sound will be stopped when the player leaves the region.
        Delay: 1 # The time to wait before stopping the sound.
      Prevent Default Sound: true # If enabled, Region Enter sound in sounds.yml won't be played.
      Sounds:
        '0':
          Delay: 0
          Options:
            Radius: 0
          Pitch: 1
          Sound: itemsadder:ambient.angelic
          Volume: 10
        '1':
          Delay: 0
          Options:
            Radius: 0
          Pitch: 1
          Sound: itemsadder:ambient.creepy
          Volume: 10
    Leave: # When a player exits this region.
      Cancellable: true
      Enabled: false
      Prevent Default Sound: true # If enabled, Region Leave sound in sounds.yml won't be played.
      Sounds:
        '0':
          Delay: 0
          Options:
            Radius: 0
          Pitch: 1
          Sound: BLOCK_NOTE_BLOCK_BASS
          Volume: 10
    Loop: # When a player enters the region, a loop will be triggered and play.
      Cancellable: true
      Delay: 0 # Time in ticks to wait to start the loop once the player enters the region.
      Enabled: false
      Period: 100 # Time in tick the loop will wait until playing the sound again.
      # If you have a long song playing, when the player leaves this region, the song
      #will be stopped instead of playing until the end. This setting applies to sounds only,
      #the loop function is stopped automatically.
      Stop On Exit:
        Delay: 10
        Enabled: true
      Prevent Other Sounds:
        Enter Sound: true # If enabled, Enter sound in regions.yml won't be played.
        Default Sound: true # If enabled, Region Enter sound in sounds.yml won't be played.
      Sounds:
        '0':
          Delay: 0
          Options:
            Radius: 0
          Pitch: 1
          Sound: itemsadder:music_disc.cdk_sunday
          Volume: 10

```

**Comment by Fargus-Van-Crystal-de-Mindark** [comment]
Prevent Other Sounds:
Enter Sound: true
this doesn't work anymore

**Comment by Fargus-Van-Crystal-de-Mindark** [comment]
If teleported in the region the sound is not activated

**Comment by Fargus-Van-Crystal-de-Mindark** [comment]
I'm on Paper 1.16.3

**Comment by Epicnicity322** [comment]
> If teleported in the region the sound is not activated

The teleport not playing is a known issue.
I have tested and the sounds are being stopped properly. Are your sounds playing at all?

**Comment by chrisrnj** [comment]
> If teleported in the region the sound is not activated

Teleporting on regions not playing sounds is a known issue.
The sounds are not being stopped because your sounds have the character '**:**'. I will fix that on the next version, but it should take a while until it is released, I suggest you rename the sound on your resource pack.

**Comment by Fargus-Van-Crystal-de-Mindark** [comment]
it is a custom name in a namespace

so I cannot rename it.

u cannot do anything?

**Comment by Epicnicity322** [comment]
You don't need namespaces on sounds, just use what you set on sounds.json on your resource pack. PlayMoreSounds automatically removes sounds with invalid characters when stopping a sound, I thought I forgot to add : but looking at the source it does not give the error when playing because the namespace is removed, but it does not remove when stopping. Even if I add : to allowed characters it will error because it is not a valid minecraft key character.

**Comment by chrisrnj** [comment]
PlayMoreSounds automatically removes invalid characters from sounds when stopping, I forgot to add : to valid characters, this check is only done on stopping sounds so that's why it is playing but not stopping. Unfortunately you will need to wait for the update for me to fix that

**Comment by chrisrnj** [comment]
You can use version 3.2.0, but I won't provide support for that.

**Comment by Fargus-Van-Crystal-de-Mindark** [comment]
I need to set a namespace, because I'm using Itemsadder 
you can check it here https://itemsadder.plugin.ga/plugin-usage/adding-content/advanced/sounds

the sound work properly with the normal /playsound in minecraft. It is completely compatible with minecraft key caracters.

anyway thank to you for your try.

I'll try to search another way to set sounds in regions


Why what is in version 3.2.0 ?

**Comment by chrisrnj** [comment]
The check for invalid characters was added on 3.3.0

**Comment by Fargus-Van-Crystal-de-Mindark** [comment]
I will try the 3.2.0 version, if it works I can use it until u have the time to add this ":" as valid characters.

**Comment by Fargus-Van-Crystal-de-Mindark** [comment]
Where is the section where it is possible to add new characters as valid? there is a folder or something similar?

**Comment by chrisrnj** [comment]
Fixed issue on commit [337f31b](https://github.com/Epicnicity322/PlayMoreSounds/commit/337f31b3047273ea86909904b263957b587e8ebc)
